From 7ecd1eee2633be97794e1eff83b32bd7e545f675 Mon Sep 17 00:00:00 2001
From: Randolph Sapp <rs@ti.com>
Date: Mon, 17 Apr 2023 15:49:27 -0500
Subject: [PATCH 256/508] arm64: dts: ti: k3-j721e-main: add gpu node

Add the IMG 8XE-GE8430 GPU device node. This cannot be upstreamed as
this device currently uses an out-of-tree kernel module.

Taken from tip of ti-linux-5.10.y branch:
991c5ce91e43cb620f534dc9fe7b0ad21f4f4388

Signed-off-by: Randolph Sapp <rs@ti.com>
Signed-off-by: Vignesh Raghavendra <vigneshr@ti.com>
---
 arch/arm64/boot/dts/ti/k3-j721e-main.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi b/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi
index f06ae0410c27..242ddf6c39a8 100644
--- a/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi
@@ -2169,6 +2169,16 @@ icssg1_mdio: mdio@32400 {
 		};
 	};
 
+	gpu: gpu@4e20000000 {
+		compatible = "ti,j721e-pvr", "img,pvr-ge8430";
+		reg = <0x4e 0x20000000 0x00 0x80000>;
+		interrupts = <GIC_SPI 24 IRQ_TYPE_LEVEL_HIGH>;
+		power-domains = <&k3_pds 125 TI_SCI_PD_EXCLUSIVE>,
+				<&k3_pds 126 TI_SCI_PD_EXCLUSIVE>;
+		power-domain-names = "gpu_0", "gpucore_0";
+		clocks = <&k3_clks 125 0>;
+	};
+
 	main_mcan0: can@2701000 {
 		compatible = "bosch,m_can";
 		reg = <0x00 0x02701000 0x00 0x200>,
-- 
2.41.0

